rollup/sync_service/sync_service.go

```diff
@@ -182,15 +194,15 @@ func (s *SyncService) fetchMessages() {
 			numMessagesPendingDbWrite = 0
 		}
 
-		s.latestProcessedBlock = lastBlock
+		s.latestProcessedBlock.Store(lastBlock)
 	}
 
 	// ticker for logging progress
 	t := time.NewTicker(LogProgressInterval)
 	numMsgsCollected := 0
 
 	// query in batches
-	for from := s.latestProcessedBlock + 1; from <= latestConfirmed; from += DefaultFetchBlockRange {
+	for from := s.latestProcessedBlock.Load() + 1; from <= latestConfirmed; from += DefaultFetchBlockRange {
```

Comment:
I am just realizing fetchMessages is a background task (ie never terminates). So, we will be calling this Load() only once at start up. Even if ResetStartSyncHeight is called, fetchMessages() will never read the new value, right?

Reply:
it will if I'm not wrong, in the case that: it finishes this round, returns and re-enters `fetchMessages`, and then it will use the updated `synced L1 height`.

and yeah, your comment here is the motivation for the `Stop + Update db + Start` solution in the previous version.

Reply:
if fetchMessages or fetchRollupEvents return (happy path or error path), in next round, they will use the new latestProcessedBlock value.

Reply:
Ok, sorry for that. I see now, it is fine :+1: